File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
        // ListNode temp = node;
        // ListNode prev = temp;
        // while (temp != null && temp.next != null){
        //     temp.val = temp.next.val;
        //     temp = temp.next;
        //     prev = temp;
        // }
        // prev.next = null;
        node.val = node.next.val;
        node.next = node.next.next;
    }
}

File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        # temp = node
        # prev = temp
        # while temp and temp.next:
        #     temp.val=temp.next.val
        #     prev = temp
        #     temp=temp.next
        # prev.next = None
        # # while node.next:
        # #     node.val=node.next.val
        # #     node=node.next
        # # node.next=None
        node.val = node.next.val
        node.next = node.next.next
        